docs/3.0.0-beta.x/guides/controllers.md

```diff
@@ -1,22 +1,151 @@
 # Controllers
 
-See the [controllers' concepts](../concepts/concepts.md#controllers) for details.
+See the [controllers' concepts](../concepts/concepts.md#controllers) for a simple overview.
 
-## How to create a controller?
+## Core controllers
+
+When you create a new Content type or a new model. You will see a new empty controller has been created. It is because Strapi builds a generic controller for your models by default and allows you to override and extend it in the generated files.
+
+### Extending a Model Controller
+
+Here are the core methods (and their current implementation).
+You can simply copy and paste this code in your own controller file to customize the methods.
+
+::: warning
+In the following example we will consider your controller, service and model is named `product`
```

Comment:
In the following example, the controller, service and model are named `product`.